ecaudit/src/test/java/com/ericsson/bss/cassandra/ecaudit/metrics/TestAuditMetrics.java

```diff
@@ -0,0 +1,116 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.ericsson.bss.cassandra.ecaudit.metrics;
+
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import com.codahale.metrics.Timer;
+import org.apache.cassandra.metrics.CassandraMetricsRegistry;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.StrictStubs.class)
+public class TestAuditMetrics
+{
+    private static final String GROUP_NAME = "com.ericsson.bss.cassandra.ecaudit";
+    private static final String METRIC_TYPE = "Audit";
+    private static final String METRIC_SCOPE = "ClientRequests";
+    private static final String METRIC_NAME_FILTER = "AuditFilter";
+    private static final String METRIC_NAME_AUDIT = "Audit";
+
+    @Mock
+    private Function<CassandraMetricsRegistry.MetricName, Timer> mockTimerFunction;
+
+    @Before
+    public void init()
+    {
+        when(mockTimerFunction.apply(any())).thenReturn(mock(Timer.class));
+    }
+
+    @Test
+    public void testFilterRequestTiming()
+    {
+        Timer mockTimer = mock(Timer.class);
+        CassandraMetricsRegistry.MetricName metric = createMetricName(METRIC_TYPE, METRIC_NAME_FILTER, METRIC_SCOPE);
+
+        when(mockTimerFunction.apply(eq(metric))).thenReturn(mockTimer);
+
+        AuditMetrics auditMetrics = new AuditMetrics(mockTimerFunction);
+        verify(mockTimerFunction, times(1)).apply(eq(metric));
+
+        auditMetrics.filterAuditRequest(999L, TimeUnit.NANOSECONDS);
+        verify(mockTimer, times(1)).update(eq(999L), eq(TimeUnit.NANOSECONDS));
+    }
+
+    @Test
+    public void testAuditRequestTiming()
+    {
+        Timer mockTimer = mock(Timer.class);
+        CassandraMetricsRegistry.MetricName metric = createMetricName(METRIC_TYPE, METRIC_NAME_AUDIT, METRIC_SCOPE);
+
+        when(mockTimerFunction.apply(eq(metric))).thenReturn(mockTimer);
+
+        AuditMetrics auditMetrics = new AuditMetrics(mockTimerFunction);
+        verify(mockTimerFunction, times(1)).apply(eq(metric));
+
+        auditMetrics.auditRequest(999L, TimeUnit.NANOSECONDS);
+        verify(mockTimer, times(1)).update(eq(999L), eq(TimeUnit.NANOSECONDS));
+    }
+
+    /**
+     * Borrowed from org.apache.cassandra.metrics.DefaultNameFactory but with tailored group name.
+     * @return a Cassandra metric name
+     */
+    private static CassandraMetricsRegistry.MetricName createMetricName(String type, String metricName, String scope)
```

Comment:
Do we really need to have the same 2 methods in the test class as well?
Maybe we can make them protected final (in AuditMetrics) so that they can be accessed from here... Also (if we need to keep them both), they can easily be tested by calling them directly...

Reply:
I could set them to package-protected final and add test cases for them instead.